ONSdigital/eq-survey-runner
# Recover State by replaying post data

### What is the context of this PR?

Fixes #364 

Questionnaire state is serialization via jsonpickle to the database. In most case that is used to restore questionnaire state. However if a deployment happens and the GIT revision changes then we replay post data, which we are now also saving, to recover the state. This solves the issue when the python classes can get out of sync between deployments.
### How to review
1. Start a survey
2. Submit at least a single page
3. Manually set the EQ_GIT_REVISION to a value other than the git head revision
4. Restart and the survey and check that the state is recovered.

Part 2 - Safe prod/pre-prod migration
1. Using master and not this branch, start a questionnaire, completing a couple of pages
2. Checkout this branch and try to restore the same session with the same questionnaire
3. It should not be able to restore the state but take you to the first page
4. A record should be included in the logs
### Who worked on the PR

@warrenbailey and @LJBabbage
